plugin/StrikeLightningConnectionStringHandler.cs

```diff
@@ -94,51 +101,12 @@ public StrikeLightningConnectionStringHandler(IServiceProvider serviceProvider,
 		if (serverUrl != null)
 			client.ServerUrl = serverUrl;
 
+		// initialize lightning client that listens on top of StrikeClient
 		var logger = _loggerFactory.CreateLogger<StrikeLightningClient>();
-
-		var connectionHash = ComputeHash(connectionString);
-		var accountFiatCurrency = GetAccountFiatCurrency(connectionHash, client, ref error);
-		if (accountFiatCurrency == null)
-			return null;
-
-		Currency targetOperatingCurrency;
-		if ("fiat".Equals(currencyStr, StringComparison.OrdinalIgnoreCase))
-		{
-			targetOperatingCurrency = accountFiatCurrency.Value;
-		}
-		else if (!Enum.TryParse(currencyStr, true, out targetOperatingCurrency))
-		{
-			error = "The key 'currency' is invalid, set either 'BTC', 'FIAT' or 'USD'/'EUR'";
-			return null;
-		}
-
-		return new StrikeLightningClient(client, db, accountFiatCurrency.Value, targetOperatingCurrency, network, logger);
-	}
-
-	private Currency? GetAccountFiatCurrency(string connectionKey, StrikeClient client, ref string? error)
-	{
-		if (_fiatCurrencyForConnection.TryGetValue(connectionKey, out var cachedCurrency))
-			return cachedCurrency;
-
-		try
-		{
-			var balances = client.Balances.GetBalances().GetAwaiter().GetResult();
-			if (!balances.IsSuccessStatusCode)
-			{
-				var errorFromServer = balances.Error?.Data;
-				error = $"The connection failed, check api key. Error: {errorFromServer?.Code} {errorFromServer?.Message}";
```

Comment:
This was primarily removed because of `GetAwaiter().GetResult()` potential for deadlocks.

You make a great point that API validation needs to be in place, so I brought back this check: https://github.com/rockstardev/strike-btcpayserver-plugin/commit/44d3105d4a178090361bb2ea7b47535f4b08787a